client/src/app/pages/applications/applications-table/applications-table.tsx

```diff
@@ -1072,11 +1066,20 @@ export const ApplicationsTable: React.FC = () => {
                               ? [
                                   {
                                     title: t("actions.analysisDetails"),
-                                    onClick: () =>
-                                      setTaskToView({
-                                        name: application.name,
-                                        task: getTask(application)?.id,
-                                      }),
+                                    onClick: () => {
+                                      const taskId = getTask(application)?.id;
+                                      taskId &&
+                                        application.id &&
```

Comment:
Minor nit-pick: I think this would look better as a simple `if (taskId && ...) { ... }`